### PR TITLE
add a file hash driver

### PR DESCRIPTION
**Pull request by SjorsO**
The new `FileHashDriver` allows you to assert that a file is identical to the file that was used to create the snapshot.

I made this PR because i'm working on a project that extracts images from .sup files, the FileHashDriver allows me to do this:
```php
/** @test */
public function it_can_extract_an_image()
{
    $sup = new Sup('example-file.sup');

    $imageFilePath = $sup->extractImage(5); // extract the 5th image in the file    

    $this->assertMatchesFileHashSnapshot($imageFilePath);
}
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ build
 composer.lock
 docs
 vendor
+.idea
```
Comment (freekmurze) [inline .gitignore]: Remove this line, it should be placed in your own global .gitignore.

File: tests/Unit/Drivers/files/example-file.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Spatie is a digital allrounder:
```
Comment (freekmurze) [inline tests/Unit/Drivers/files/example-file.txt]: ❤️

File: .gitignore (hunk 1 above)
Comment (SjorsO) [reply .gitignore]: done

**Comment by SjorsO** [comment]
@freekmurze I reverted the gitignore change, is there anything else you would like changed?

**Comment by freekmurze** [comment]
For me this is ok, waiting for the final check of @sebastiandedeyne

File: src/MatchesSnapshots.php [hunk 3]
```diff
@@ -36,6 +38,17 @@ public function assertMatchesJsonSnapshot($actual)
         $this->assertMatchesSnapshot($actual, new JsonDriver());
     }
 
+    public function assertMatchesFileHashSnapshot($filePath)
+    {
+        if (! file_exists($filePath)) {
+            throw new Exception('File does not exist');
```
Comment (sebastiandedeyne) [inline src/MatchesSnapshots.php]: Would it be possible to fail the test here instead?

**Comment by SjorsO** [comment]
@sebastiandedeyne i've applied the change you requested

**Comment by sebastiandedeyne** [comment]
Thanks!

**Pull request merged**